### PR TITLE
DealProposal Label type switch

### PR DESCRIPTION
**Pull request by laudiacay**
For issue #187

### CODE REVIEW

**Comment by laudiacay** [comment]
Hi- this is my first FIP, and I think it sounds very silly, but I wasn't sure what to put in some of the fields because this is such a tiny change. Please let me know what you think.

File: FIPS/fip-dealproposal-label-string-to-bytes.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Design Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+Design decisions are detailed extensively in above sections.
+
+## Backwards Compatibility
+<!--All FIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The FIP must explain how the author proposes to deal with these incompatibilities. FIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+
+This changes specs-actors. It should be backwards-compatible, in that the new version will accept any sort of String from the previous version. However, out-of-date clients that validate UTF-8 on this field may reject newly-valid non-UTF-8 labels.
```
Comment (ZenGround0) [inline FIPS/fip-dealproposal-label-string-to-bytes.md:45]: To reduce complexity all previous label fields should be moved from cbor strings to cbor bytes.  This accomplishes accepting all previous strings without introducing versioning of proposal data.

This change requires a migration which we'll need to describe here.  Specifically all deal proposals on chain will need to change the encoding of the label field.

File: FIPS/fip-dealproposal-label-string-to-bytes.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for FIPs that are affecting consensus changes. Other FIPs can choose to include links to test cases if applicable.-->
+
+Checking that a non-UTF-8 string is accepted as a DealProposal Label will be sufficient to test this FIP.
```
Comment (ZenGround0) [inline FIPS/fip-dealproposal-label-string-to-bytes.md:50]: I think the problem is that non-UTF-8 strings are accepted, and this breaks the cbor spec.  

To test this FIP I think we just need to check that v0-v6 proposals labels become cbor bytes with the same raw byte payload after migration, and that new proposals have labels encoded as cbor bytes.

File: FIPS/fip-dealproposal-label-string-to-bytes.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+created: 2021-09-29
+spec-sections: 
+  - specs-actors
+---
+
+<!--You can leave these HTML comments in your merged FIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new FIPs. Note that a FIP number will be assigned by an editor. When opening a pull request to submit your FIP, please use an abbreviated title in the filename, `fip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the FIP.-->
```
Comment (ZenGround0) [inline FIPS/fip-dealproposal-label-string-to-bytes.md:18]: It's worth mentioning here or in the abstract that the proposal struct is technically invalid cbor with the acceptance of non-UTF-8 cbor strings.

File: FIPS/fip-dealproposal-label-string-to-bytes.md (hunk 1 above)
Comment (laudiacay) [reply FIPS/fip-dealproposal-label-string-to-bytes.md]: is the current change correct? do I write the migration in specs-actors?

**Comment by laudiacay** [comment]
TODO want to also add a bound on the length of the label field- this is not ready to go.

**Comment by laudiacay** [comment]
> TODO want to also add a bound on the length of the label field- this is not ready to go.

never mind this was already done lol. just need to write the migration and this will be ready :)

**Comment by laudiacay** [comment]
Migration done. I think this is ready to go? https://github.com/filecoin-project/specs-actors/pull/1496

**Review by ZenGround0 — APPROVED** [review]

File: FIPS/fip-dealprop-label-type.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+---
+fip: <to-be-assigned>
```
Comment (kaitlin-beegle) [inline FIPS/fip-dealprop-label-type.md:2]: FIP0027! Let's add it in, and we'll be ready for 'Last Call' status! :)

**Comment by anorth** [comment]
@kaitlin-beegle and FIP editors, can we please merge this? Leave it in Draft if you want to align Last Call status with imminent implementation (though I would just go to Last Call). But leaving the PR open is no-one's ideal solution.

**Review by kaitlin-beegle — COMMENTED** [review]
Just updated FIP status and FIP number; ready to merge.

**Pull request merged**